enable_disable_resources.md

```diff
@@ -0,0 +1,366 @@
+resource is disabled.
+
+Here's what that might look like:
+
+```go
+type KubernetesApplySpec struct {
+  ...
+  
+  Disabled *bool
+  DisableSource *DisableSource
+}
+
+type KubernetesApplyStatus struct {
+  ...
+  
+  DisableStatus *DisableStatus
+}
+
+type DisableSource struct {
+  ConfigMap *ConfigMapDisableSource
+}
+
+type ConfigMapDisableSource struct {
+  // The name of the ConfigMap
+  Name string
+    
+  // The key where the enable/disable state is stored.
+  Key string
+  
+  // Specify whether the ConfigMap must be defined.
+  Optional *bool
+}
+
+type DisableStatus struct {
+  Disabled bool
+  LastUpdateTime metav1.Time
+  Reason string
+}
+```
+
+The DisableStatus for a set of objects would be aggregated into UIResource like
+in the previous proposal.
+
+There's some prior art here: I've borrowed this pattern from Kubernetes, where
+you can specify the environment variables for a container with `Env` and `EnvFrom`.
+
+https://pkg.go.dev/k8s.io/api/core/v1#Container
+
+You could imagine starting out by just implementing one of the fields
+(either `Disabled` or `DisableSource` would be fine).
+
+For the initial implementation, I would have the Tiltfile create ConfigMaps for
+each UIResource, and wire up the each object's DisableSource to the correct
+ConfigMap.
+
+The downside of this approach is that, if you want the UI to be able to disable a resource,
+what API does it call and how does it know how to call it? Does it follow DisableSource
+to the ConfigMap, then update the ConfigMap? Or does it support a few common conventions?
+
+I could imagine a lot of complexity on the UI side to figure out which property
+to update for each type of object.
+
+### Resource Selector
+
+We add a brand new type of object to our system: `Resource`. A `Resource` groups
+together a set of objects by label. The closest analogue in the Kubernetes ecosystem
+is a Service or an Ingress.
+
+```go
+type ResourceSpec struct {
+  Selectors []ResourceSelector
+  
+  Disabled bool
+}
+
+// Selects a group of resources
+type ResourceSelector struct {
+  Group string
+  Version string
+  Kind string
+  
+  // +optional
+  Labels map[string]string
+  
+  // +optional
+  Name string
+}
+```
+
+There's a direct 1:1 correspondence between a `Resource` model and the
+`UIResource` that exists today.
+
+A variant on this proposal is that we merge `Resource` and `UIResource` into a
+single object where the Spec has the selector and disable state, and the Status
+has all the aggregate status of those objects.
+
+When the UI wants to disable a Resource, it updates the spec of the Resource
+object.
+
+If the Resource is disabled, the Resource controller watches for objects that
+match its selector, and immediately deletes them whenever they're created.
+
+One interesting property of this proposal is individual API object controllers
+don't need to implement disable functionality. If they handle deletes properly,
+then a delete is equivalent to a disable.
```

Comment:
IIUC
* In our current implementation, we'd implement this by creating a ResourceSpec that points at all of a Tiltfile's children, but not their children, so basically just the `toApiObjects` results (probably minus the UIResource, so we can still show it in the UI)
* In the long run, maybe a Tiltfile just creates Resource objects, and the Resource reconciler creates the KubernetesApplys, Cmds, etc, and the Resource reconciler can just skip creating the objects in the first place? (this is still all pretty vague to me)

Reply:
1) ya
2) i generally think we want to avoid having a Resource object that contains templates for all objects in tilt